nucypher/blockchain/eth/actors.py

```diff
@@ -764,6 +764,9 @@ def _prolong_stake(self, stake_index: int, lock_periods: int) -> TxReceipt:
 
     def _deposit(self, amount: int, lock_periods: int) -> TxReceipt:
         """Public facing method for token locking."""
+        self.token_agent.approve_transfer(amount=0,
+                                          spender_address=self.staking_agent.contract_address,
+                                          transacting_power=self.transacting_power)
```

Comment:
Because `approve_and_call` requires allowance to be 0, we either have to keep this horrific workaround

Reply:
in `approve_and_call` on contract side we can rewrite only `call` part because `approve` part happens inside token contract (which is not upgradeable). So in normal case allowance is already zero and first tx should be skipped

Reply:
What happens if we have multiple `approveAndCall` method calls? Assuming allowance is zero, the first call will work and thus set allowance to some amount larger than 0. Now, because the allowance is non-zero, the second call will fail. I think this is what happened in the issue described by @KPrasch 
>Yes, I agree - as demonstrated in a discord discussion this can also happen while trying to create a new stake with an account that has previously created stakes. 
I updated the code to handle this case.

Reply:
after executing `approveAndCall` allowance will be zero again, so it's not zero only when something fail